automatically_annotate_mass2motifs/bin_spectra.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+from typing import List
+from tqdm import tqdm
+from matchms import Spectrum, Fragments
+from matchms.filtering import add_losses
+from speed_up_ms2lda.utils import convert_file_to_matchms_spectrum_objects
+
+
+class Binner():
+    """Class for creating bins and storing settings"""
+    def __init__(self, bin_width):
+        self.bin_width = bin_width
+        self.number_of_decimals = len(str(bin_width).split(".")[1])
+
+    def return_mass_bin(self, mass):
+        mass_bin = mass - mass % self.bin_width + 0.5 * self.bin_width
+        return round(mass_bin, self.number_of_decimals + 1)
+
+    def return_binned_masses(self, list_of_masses, list_of_intensities):
+        binned_masses = {}
+        for i, mass in enumerate(list_of_masses):
+            mass_bin = self.return_mass_bin(mass)
+            intensity = list_of_intensities[i]
+            if mass_bin in binned_masses:
+                binned_masses[mass_bin] += intensity
```

Comment:
Seems fine. But just fyi: This is a different binning algorithm than MS2DeepScore which would take the maximum intensity per bin. Here it will take the sum of all intensities per bin.

Reply:
Not saying one is better than the other...
If you wanted to change you could use a `=max(intensity, binned_masses[mass_bin])`.